python/pyspark/sql/catalog.py

```diff
@@ -133,6 +133,35 @@ def listColumns(self, tableName, dbName=None):
                 isBucket=jcolumn.isBucket()))
         return columns
 
+    def tableExists(self, tableName, dbName=None):
+        """Check if the table or view with the specified name exists.
+        This can either be a temporary view or a table/view.
+
+        .. versionadded:: 3.3.0
+
+        Parameters
+        ----------
+        tableName : string
+                    name of the table to check existence
+        dbName : string, optional
+                 name of the database to check table existence in.
+                 If no database is specified, the current database is used
+
+        Returns
+        -------
+        Boolean indicating whether the table/view exist
+
+        Examples
+        --------
+        >>> spark.catalog.tableExists("unexisting_table")
```

Comment:
The testing framework actually run these examples as tests. So the table would have to be created (and dropped to clean up)

Reply:
Fixed